src/scenarios/shared/runner.py

```diff
@@ -9,38 +9,69 @@
 from collections import namedtuple
 from argparse import ArgumentParser
 from shared.startup import StartupWrapper
-from shared.util import publishedexe, extension
+from shared.util import publishedexe, extension, pythoncommand, iswin
 from shared import const
 from performance.logger import setup_loggers
 
-
-reqfields = ('exename',
-            )
-optfields = ('guiapp',
-             'startupmetric',
-             'appargs',
-             'environmentvariables',
-             'iterations',
-             'timeout',
-             'warmup',
-             'workingdir',
-             'iterationsetup',
-             'setupargs',
-             'iterationcleanup',
-             'cleanupargs',
-             'processwillexit',
-             'measurementdelay'
-             )
-
 # These are the kinds of scenarios we run. Default here indicates whether ALL
 # scenarios should try and run a given test type.
 testtypes = {const.STARTUP: False,
              const.SDK: False,
              const.CROSSGEN: False}
 
-TestTraits = namedtuple('TestTraits',
-                        reqfields  + tuple(testtypes.keys()) + optfields,
-                        defaults=tuple(testtypes.values()) + (None,) * len(optfields))
+
+class TestTraits:
+    traits = {}
+    # Scenario-independent traits that can be set in test.py,
+    # but can also be overriden in scenario-specific case
+    reqfields = ('exename',
+                 )
+    optfields = ('guiapp',
+                 'startupmetric',
+                 'appargs',
+                 'iterations',
+                 'timeout',
+                 'warmup',
+                 'workingdir',
+                 'iterationsetup',
+                 'setupargs',
+                 'iterationcleanup',
+                 'cleanupargs',
+                 'processwillexit',
+                 'measurementdelay',
+                 'environmentvariables'
+                 )
```

Comment:
I would make these part of the class that you can fill in later. Put differently: now that `TestTraits` is a class it really is a class with the semantic that it holds some required fields and some optional fields. Make the class do that, and then runner can set it up as it sees fit.

Reply:
I also thought about having the traits as fields of the class but it seems not as flexible as just having a dictionary ```traits={ }```. So if we make them class fields I assume it will 
1) call ```setattr(traits_object, key, kwargs[key])``` to assign value to the field if the field name is a variable
2) call ```vars(traits_object)``` to get all the trait fields 

Another concern is it'd be very easy to do ```trait_object.field=<some random value>``` (but easier access as trade-off)